rpc_server.go

```diff
@@ -261,14 +267,28 @@ func (s *SourceControl) Stop(dummy *string, reply *bool) error {
 	}
 	log.Printf("Stopping data source\n")
 	s.activeSource.Stop()
-	s.status.Running = false
-	s.activeSource = nil
+	s.observeSourceStop()
 	*reply = true
 	s.broadcastStatus()
 	*reply = true
 	return nil
 }
 
+// observeSourceStop modifies s to make future messag
```

Comment:
Is this comment complete? Well, obviously it isn't.

Reply:
fixed